platform-tests/src/test/java/org/junit/platform/console/tasks/XmlReportAssertions.java

```diff
@@ -26,24 +28,27 @@
  */
 class XmlReportAssertions {
 
-	private static Validator schemaValidator;
+	private static AtomicReference<Schema> schema = new AtomicReference<>();
```

Comment:
Do you really need an `AtomicReference` why not just use `getSchema` as initializer?

Reply:
It's surprisingly hard to implement a thread-safe singleton without locking. If you have a better solution, I'm open to change it. 😉

Reply:
Did someone say _thread-safe singleton_? :)

Would Effective Java's recommended solution - a single-instance enum - work in this case? E.g.

```java
enum ClassName extends AppropriateInterface {
    INSTANCE;

   // needed methods go here
}
```

Reply:
I had tried that at some point but it felt clunky. Revisiting it now, it's not actually that bad so I changed it in 135a07d7ce28de1605bb69fbd2e24e9cdf8fc689.

@jbduncan Thanks! 🙂

Reply:
@marcphilipp You're very welcome! :smiley: